interfaces/repo.go

```diff
@@ -322,68 +322,100 @@ func (r *Repository) Connect(plugSnapName, plugName, slotSnapName, slotName stri
 //   from all the slots found therein. It is not an error if there are no
 //   such plugs but it is still an error if the snap does not exist or has no
 //   slots at all.
-func (r *Repository) Disconnect(plugSnapName, plugName, slotSnapName, slotName string) error {
+//
+// Disconnect returns the list of severed connections, the list of affected snaps
+// and an error if one occurred.
+func (r *Repository) Disconnect(plugSnapName, plugName, slotSnapName, slotName string) ([]ConnRef, []*snap.Info, error) {
 	r.m.Lock()
 	defer r.m.Unlock()
 
+	var conns []ConnRef
+	var snaps map[string]*snap.Info
+	var err error
+
 	switch {
-	case plugSnapName == "" && plugName == "" && slotName == "":
-		// Disconnect everything from slotSnapName
-		return r.disconnectEverythingFromSnap(slotSnapName)
```

Comment:
Do we really don't need that case anymore?  Isn't that required internally, when removing the snap?

As we discussed, we should not expose that in the CLI, but it's not clear to me if that's not used at all internally.